File: ITP1/04/4_d.py
n = int(input())
l = list(map(int, input().split()))

min = min(l)
max = max(l)
sum = sum(l)

print(f"{min} {max} {sum}")

"""
別解１
BIG_NUM = 2000000000

N = int(input())

table = list(map(int,input().split()))

maximum = -BIG_NUM
minimum = BIG_NUM
total = 0

for x in table:
    if maximum < x:
        maximum = x
    if minimum > x:
        minimum = x
    total += x

print("%d %d %d"%(minimum,maximum,total))
"""

File: ALDS1/c++/04/4_d.cpp
#include <iostream>
using namespace std;

int main()
{
    int n, min = 10000005, max = -10000005;
    long long sum = 0;
    cin >> n;
    int num[n];
    for (int i = 0; i < n; i++)
    {
        cin >> num[i];
    }
    for (int i = 0; i < n; i++)
    {
        if (min > num[i])
            min = num[i];
        if (max < num[i])
            max = num[i];
        sum += num[i];
    }
    cout << min << " " << max << " " << sum << endl;

    return 0;
}